nengo_loihi/version.py

```diff
@@ -5,10 +5,40 @@
 '.devN' will be added to the version unless the code base represents
 a release version. Release versions are git tagged with the version.
 """
+import warnings
+
+
+def check_nengo_version():
+    import nengo
+
+    if nengo.version.version_info < minimum_nengo_version_info:
+        raise ValueError(
+            "nengo-loihi does not support Nengo version %s. "
+            "Upgrade with 'pip install --upgrade --no-deps nengo'."
+            % nengo.__version__)
+    elif nengo.version.version_info > latest_nengo_version_info:
+        warnings.warn("This version of `nengo_loihi` has not been tested "
+                      "with your `nengo` version (%s). The latest fully "
+                      "supported version is %s" % (
+                          nengo.__version__, latest_nengo_version))
+
+
+def info2string(info):
+    return '.'.join(str(v) for v in info)
+
 
 name = "nengo_loihi"
 version_info = (0, 7, 0)  # (major, minor, patch)
 dev = 0
 
-version = "{v}{dev}".format(v='.'.join(str(v) for v in version_info),
+version = "{v}{dev}".format(v=info2string(version_info),
                             dev=('.dev%d' % dev) if dev is not None else '')
+
+# --- Nengo version compatibility
+# oldest nengo version we are compatible with (test on release)
+minimum_nengo_version_info = (2, 7, 0)
+minimum_nengo_version = info2string(minimum_nengo_version_info)
+
+# newest nengo version we are compatible with (update on release)
+latest_nengo_version_info = (2, 8, 0)
```

Comment:
The way I do it in NengoDL is that for dev versions I give the nengo dev version compatibility (which would be 3.0.0 in this case).  Because we are testing that, so they should be compatible, and it's also a pain getting this warning all the time when you're using dev versions.  Then when you do a release, you downgrade this to the latest supported nengo release (e.g. 2.8.0 if we were to do a release now).

Then I have this test https://github.com/nengo/nengo-dl/blob/master/nengo_dl/tests/test_version.py#L16 to make sure that we don't forget to update the version when doing a release.

I'd suggest that we do the same for nengo-loihi.

Reply:
I think that that makes sense.